master/internal/api_checkpoint.go

```diff
@@ -28,6 +29,42 @@ func (a *apiServer) GetCheckpoint(
 	}
 }
 
+func (a *apiServer) CreateCheckpoint(
+	_ context.Context, req *apiv1.CreateCheckpointRequest,
+) (*apiv1.CreateCheckpointResponse, error) {
+	log.Infof("adding checkpoint %s (trial %d, batch %d)",
```

Comment:
I would change these to `Debugf` before landing or make a note to do it after a release or two if you want them for debugging

Reply:
it is not for debugging. We will be removing checkpointing workload in favor of logging for checkpointing itself.

Reply:
let's talk about this, I was unaware we were removing checkpoint workloads.